app/controllers/feeds_controller.rb

```diff
@@ -1,15 +1,36 @@
 class FeedsController < ApplicationController
   respond_to :html, :atom
+
   def index
+    urls = User.pluck(:feed_url).reject { |f| f.blank?}
+    hydra = Typhoeus::Hydra.new
+    requests = urls.map do |url|
+      req = Typhoeus::Request.new(url, followlocation: true)
+      hydra.queue(req)
+      req
+    end
+    hydra.run
+
     @entries = []
-    urls = User.pluck(:feed_url).reject {|f| f.blank? }
-    @feeds = Feedjira::Feed.fetch_and_parse(urls)
-    @feeds.each do |url, feed|
-      if feed.respond_to?(:entries) && feed.entries.present?
-        feed.entries.each {|e| @entries << e }
+    requests.each do |req|
+      begin
+        feed = Feedjira::Feed.parse(req.response.options[:response_body])
+        if feed.class == Feedjira::Parser::Atom
+          feed.entries.each do |entry|
+            @entries << entry
+          end
+        elsif feed.class == Feedjira::Parser::RSS
+          feed.entries.each do |entry|
+            @entries << entry
+          end
+        else
+          raise "Unknown feed format."
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.